File: Activity/Week2/Python/perform-string-shifts.py
class Solution:
    def stringShift(self, s: str, shift: List[List[int]]) -> str:
        strLen = len(s)
        total = 0
        
        for i in range(len(shift)):
            direction = shift[i][0]
            amount = shift[i][1]
            
            amount %= strLen;
            if direction == 1:
                amount = strLen - amount
                
            total += amount
            
        total %= strLen
        s = list(s)
        for _ in range(total):
            temp = s[0]
            for i in range(strLen-1):
                s[i] = s[i + 1]
                
            s[strLen - 1] = temp
        
        return "".join(s)

File: Explore/2020April/Week2/C/perform-string-shifts.c
char * stringShift(char * s, int** shift, int shiftSize, int* shiftColSize){
    int len = strlen(s);
    int total = 0;
    
    for(int i = 0; i < shiftSize; i++)
    {
        int direction = shift[i][0];
        int amount = shift[i][1];
        
        amount =  amount % len;
        if(direction == 1)
            amount = len - amount;
        
        total += amount;
    }
    
    total %= len;
    
    for(int i = 0; i < total; i++)
    {
        char temp = s[0];
        for(int j = 0; j < len - 1; j++)
            s[j] = s[j + 1];
        
        s[len - 1] = temp;
    }
    
    return s;
}
